src/nc_connection.c

```diff
@@ -341,7 +345,12 @@ conn_recv(struct conn *conn, void *buf, size_t size)
     ASSERT(conn->recv_ready);
 
     for (;;) {
-        n = nc_read(conn->sd, buf, size);
+        if (conn->ssl != NULL) {
```

Comment:
Seems like this pattern is everywhere - do you think it's worth being all like: `n = nc_read_from(conn, buf, size)` and then all like:
```
ssize_t nc_read_from(struct conn *conn, void *buf, size_t size)
{
    return conn->ssl ?
        nc_ssl_read(conn->ssl, buf, size) :
        nc_read(conn->sd, buf, size);
}
```

Reply:
This is the only place we read so a function feels overkill, but I like the inline ternary since that's a bit cleaner.